rllib/core/models/catalog.py

```diff
@@ -0,0 +1,227 @@
+        fcnet_hiddens = model_config["fcnet_hiddens"]
+        encoder_latent_dim = model_config["encoder_latent_dim"] or fcnet_hiddens[-1]
+
+        if model_config["use_lstm"]:
+            encoder_config = LSTMEncoderConfig(
+                input_dim=observation_space.shape[0],
+                hidden_dim=model_config["lstm_cell_size"],
+                batch_first=not model_config["_time_major"],
+                num_layers=1,
+                output_dim=model_config["lstm_cell_size"],
+                output_activation=output_activation,
+                observation_space=observation_space,
+                action_space=action_space,
+                view_requirements_dict=view_requirements,
+                get_tokenizer_encoder_config=self.get_tokenizer_encoder_config,
+            )
+        elif model_config["use_attention"]:
+            raise NotImplementedError
+        else:
+            # TODO (Artur): Maybe check for original spaces here
+            # input_space is a 1D Box
+            if isinstance(observation_space, Box) and len(observation_space.shape) == 1:
+                # In order to guarantee backward compatability with old configs,
+                # we need to check if no latent dim was set and simply reuse the last
+                # fcnet hidden dim for that purpose.
+                if model_config["encoder_latent_dim"]:
+                    hidden_layer_dims = model_config["fcnet_hiddens"]
+                else:
+                    hidden_layer_dims = model_config["fcnet_hiddens"][:-1]
+                encoder_config = MLPEncoderConfig(
+                    input_dim=observation_space.shape[0],
+                    hidden_layer_dims=hidden_layer_dims,
+                    hidden_layer_activation=activation,
+                    output_dim=encoder_latent_dim,
+                    output_activation=output_activation,
+                )
+
+            # input_space is a 3D Box
+            elif (
+                isinstance(observation_space, Box) and len(observation_space.shape) == 3
+            ):
+                if not model_config.get("conv_filters"):
+                    model_config["conv_filters"] = get_filter_config(
+                        observation_space.shape
+                    )
+
+                encoder_config = CNNEncoderConfig(
+                    input_dims=observation_space.shape,
+                    filter_specifiers=model_config["conv_filters"],
+                    filter_layer_activation=activation,
+                    output_activation=output_activation,
+                    output_dim=encoder_latent_dim,
+                )
+            # input_space is a possibly nested structure of spaces.
+            else:
+                # NestedModelConfig
+                raise NotImplementedError("No default config for complex spaces yet!")
+
+        return encoder_config
+
+    @classmethod
+    def get_tokenizer_encoder_config(
```

Comment:
I thought about using other, specialised, models here as tokenizers.
But that would lead to a situation where everything we currently have as Encoders would be duplicated.
For example we'd have a CNNTokenizer (or CNNTrunk + flatten) that would be the same as the CNNEncoder but lacking the state pass-through. I think we should reuse the code here because these encoders are essentially a of what the tokenizers would be.